src/flatcollections-array/FlatArray/FlatArray.T/Functional/Functional.Chunk.cs

```diff
@@ -0,0 +1,40 @@
+namespace System;
+
+partial struct FlatArray<T>
+{
+    public FlatArray<FlatArray<T>> Chunk(int size)
+    {
+        if (size < 1)
+        {
+            throw InnerExceptionFactory.SizeOutsideBounds(nameof(size), size);
+        }
+
+        if (length == default)
+        {
+            return default;
+        }
+
+        var chunksLength = length / size;
+        if (length % size != default)
+        {
+            chunksLength++;
+        }
+
+        var chunks = new FlatArray<T>[chunksLength];
+        for (var i = 0; i < chunks.Length; i++)
+        {
+            var start = i * size;
+
+            var chunkLength = length - start;
+            if (chunkLength > size)
+            {
+                chunkLength = size;
+            }
+
+            var chunkItems = InnerArrayHelper.CopySegment(items!, start, chunkLength);
+            chunks[i] = new(chunkLength, chunkItems);

```

Comment:
Consider using this private constructor targeted for creating FlatArray with the same length as the underlying array.

```suggestion
            chunks[i] = new(chunkItems, default);
```

Reply:
Fixed